src/tests/simplekdf.rs

```diff
@@ -70,8 +91,65 @@ fn test_concatenate_kdf() {
     assert_eq!(check_object_validation(session, dk_handle, 0), true);
 
     let exp_value = hex::decode("0123456789abcdef").unwrap();
-    let value = ret_or_panic!(extract_key_value(session, dk_handle));
-    assert_eq!(value, exp_value);
+    if let Some(err) = check_attributes(
+        session,
+        dk_handle,
+        &[(CKA_VALUE_LEN, 8)],
+        &[(CKA_VALUE, &exp_value)],
+        &[
+            (CKA_TOKEN, token),
+            (CKA_EXTRACTABLE, true),
+            (CKA_SENSITIVE, false),
+            (CKA_NEVER_EXTRACTABLE, false),
+            //(CKA_ALWAYS_SENSITIVE, false),
+            (CKA_ALWAYS_SENSITIVE, true), // FIXME should be false
```

Comment:
what's on the original key?
Is it possible we are not setting CKA_ALWAYS_SENSITIVE at all on creation?
That could result on the default (true) being returned in internal queries (it would also be a bug probably).

Reply:
I was trying to check the attributes in the new test below, `test_concatenate_kdf_attributes()` and the attributes on the base key looked sensible. But it might be that they were derived from the `CKA_SENSITIVE` on the fly. Let me double-check they are created correctly.

Reply:
Just mentioning that we discuss this offline and we have a path forward that differs from the comment here. We will ensure importing an object also stores the the attributes properly so these attributes are always available on all key objects.

Reply:
The current tests should be working with the current version setting the default attributes according to the specs.
What is not done is setting the default `CKA_ALWAYS_SENSITIVE` and `CKA_NEVER_EXTRACTABLE` for private keys and other secret keys (AES, HMACs ...), which requires the same semantics as we do not have now separate common create method now.
I can try to move them all into the `default_object_create()`, but then I need to distinguish the object class in there (should be in the template, I think) or create separate `private_key_object_create()`. I guess I will try the first approach as it makes the change in one place, which is easily traceable to the PKCS#11 specs

Reply:
Done in the current iteration